Reduce complexity of string comparisons

uyjulian opened the pull request:
As per the following article: https://nee.lv/2021/02/28/How-I-cut-GTA-Online-loading-times-by-70/

These changes will reduce the complexity of string comparisons, increasing performance in most cases:
* Using `TJSMapGlobalStringMap` where possible in order to use a single pointer for comparison instead of comparing a buffer
* Using `tTJSString::GetHint` where possible to reduce the time it takes to search the hash map
* Avoiding the use of `TJS_strcmp` which may introduce quadratic complexity
* Add direct pointer comparison of `const tjs_char *` to `TJS_strnicmp`, `TJS_stricmp`, `TJS_strcmp`, and `TJS_strncmp` (decreasing the string comparison complexity in most cases)

uyjulian committed:
Add static tTJSString variables to global string map

uyjulian committed:
Use string hint for iTJSDispatch2::{DeleteMember,Operation,PropGet,PropSet} whenever possible

uyjulian committed:
Avoid using TJS_strcmp for string comparison when possible

Akarinnnnn commented:
Seems good, since those names in code are all basic latin, they don't have to face various encodings and cultural specific variants .

uyjulian commented:
> Seems good, since those names in code are all basic latin, they don't have to face various encodings and cultural specific variants .

This only relates to the pointer of the string. This does not relate to the contents of the string.

uyjulian committed:
Check if pointers are same in TJS_strnicmp / TJS_stricmp / TJS_strcmp / TJS_strncmp

uyjulian commented:
I replaced the explicit pointer comparison in tTJSString and moved them to `TJS_strnicmp`, `TJS_stricmp`, `TJS_strcmp`, and `TJS_strncmp` instead. This should make them more useful there
